### PR TITLE
feat: add --monthly flag to committers

### PR DESCRIPTION
**Pull request by marco-ippolito**

### CODE REVIEW

File: libs/report/committers/src/lib/parse-monthly.ts [hunk 1]
```diff
@@ -48,3 +33,20 @@ export function parseMonthly(
...
+  return {
+    ...data,
+    start: format(data.start, 'yyyy-MM-dd'),
+    end: format(data.end, 'yyyy-MM-dd'),
+  };
+}
+
+export function outputMonthlyCommitters(commits: MonthlyData[]) {
+  const mapped = commits.map((c) => monthlyDataHumanReadable(c));
+  console.table(mapped);
```
Comment (gkalpak) [inline libs/report/committers/src/lib/parse-monthly.ts:47]: The committers object will not look great as a table cell value. Should we format it separately or maybe just display the total number of commits per month?

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: libs/report/committers/src/lib/committers.ts [hunk 2]
```diff
@@ -6,13 +6,19 @@ import { parseGitLogEntries } from './parse-git-log-entries';
 import { getCommitterCounts } from './get-committer-counts';
 import { ArgumentsCamelCase, CommandBuilder, CommandModule } from 'yargs';
 import { CommitterCount } from './types';
+import {
+  outputMonthlyCommitters,
+  outputMonthlyCommittersJson,
+  parseMonthly,
+} from './parse-monthly';
```
[the remark this answers is not in the document]
Comment (marco-ippolito) [reply libs/report/committers/src/lib/committers.ts:13]: I wish nx format re-ordered them 😬

File: libs/report/committers/src/lib/parse-monthly.ts [hunk 3]
```diff
@@ -1,44 +1,29 @@
...
       continue;
     }
     const [start, end] = [range[idx], range[idx + 1]];
     const month: MonthlyData = {
-      name: format(start, 'LLLL yyyy'),
-      start,
-      end,
+      month: format(start, 'LLLL yyyy'),
+      start: startDate,
+      end: endDate,
```
[the remark this answers is not in the document]
Comment (marco-ippolito) [reply libs/report/committers/src/lib/parse-monthly.ts:16]: Not sure what you mean? Like the start of the month and end of the month?

[the remark this answers is not in the document]
Comment (gkalpak) [reply libs/report/committers/src/lib/parse-monthly.ts:16]: For most months yes. Except for the beginning/end of the period, which might be a partial month.
For example, if the overall range is 2024-02-15 to 2024-04-15, I would expect monthly entries with ranges 2024-02-15 to 2024-03-01, 2024-03-01 to 2024-04-01, 2024-04-01 to 2024-04-15 (or something like that).

[the remark this answers is not in the document]
Comment (marco-ippolito) [reply libs/report/committers/src/lib/parse-monthly.ts]: done

File: libs/report/committers/src/lib/parse-monthly.ts (hunk 1 above)
Comment (gkalpak) [reply libs/report/committers/src/lib/parse-monthly.ts:45]: Sounds good to me :+1:

**Review by aavorthmann — APPROVED** [review]

File: libs/report/committers/src/lib/parse-monthly.ts [hunk 4]
```diff
@@ -1,44 +1,29 @@
...
     };
 
     for (const rec of entries) {
-      if (isWithinInterval(new Date(rec.date), { start, end })) {
-        month.committers[rec.committer] = month.committers[rec.committer] || [];
-        month.committers[rec.committer].push({
-          hash: rec.commitHash,
-          date: rec.date,
-        });
+      if (isWithinInterval(rec.date, { start, end })) {
```
Comment (gkalpak) [inline libs/report/committers/src/lib/parse-monthly.ts:25]: I don't think this is correct. The entry's `rec.date` has been retrieved by git based on the unaltered start-/end-dates. Here, we are comparing them with the modifed `start`/`end` dates. Some commits could get misaccounted for.

File: libs/report/committers/src/lib/parse-monthly.ts [hunk 5]
```diff
@@ -1,44 +1,29 @@
...
+    start: startDate,
+    end: endDate,
+  });
+
+  for (let idx = 0; idx < range.length - 1; idx++) {
+    // `convertToUTC` is an awful hack to convert the date to UTC
+    // This is needed because the date-fns eachMonthOfInterval function
+    // is not timezone aware and will return the date in the local timezone
+    // Refs: https://github.com/date-fns/date-fns/issues/3097
+    const [start, end] = [convertToUTC(range[idx]), convertToUTC(range[idx + 1])];
```
Comment (gkalpak) [inline libs/report/committers/src/lib/parse-monthly.ts:16]: I don't think it is right to do this `convertToUTC()` transformation, unless we also do it to the initial before-/after-dates that we pass to the git command.

Everywere else, we accept that we are dealing with dates that represent start/end of periods in the user's timezone. We should either stick to that or convert the dates from the beginning to represent the start/end of periods in UTC.

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: libs/report/committers/src/lib/parse-monthly.ts (hunk 5 above)
Comment (marco-ippolito) [reply libs/report/committers/src/lib/parse-monthly.ts:16]: There is no timezone where the end of the month start at `11pm` thats the issue 😄 so I don't think its acceptable

File: libs/report/committers/src/lib/parse-monthly.ts (hunk 4 above)
Comment (marco-ippolito) [reply libs/report/committers/src/lib/parse-monthly.ts:25]: Not sure what you mean? We are checking if commit date X is between date Y and Z

**Review by gkalpak — APPROVED** [review]
LGTM with the addition of https://github.com/marco-ippolito/herodevs-cli/pull/1 :+1:

**Pull request merged**